src/util.js

```diff
@@ -55,29 +25,42 @@ exports.serialize = (dagNode, callback) => {
   callback(null, serialized)
 }
 
-exports.deserialize = (data, callback) => {
+function deserialize (data, callback) {
   const pbn = proto.PBNode.decode(data)
 
   const links = pbn.Links.map((link) => {
     return new DAGLink(link.Name, link.Tsize, link.Hash)
   })
 
-  stable.inplace(links, exports.linkSort)
-
   const buf = pbn.Data || new Buffer(0)
 
-  const dagNode = new DAGNode(buf, links)
-
-  callback(null, dagNode)
+  DAGNode.create(buf, links, callback)
 }
 
-exports.cid = (dagNode, callback) => {
-  dagNode.multihash((err, multihash) => {
-    if (err) {
-      return callback(err)
-    }
-    const cid = new CID(multihash)
-    callback(null, cid)
-  })
+function toProtoBuf (node) {
+  const pbn = {}
+
+  if (node.data && node.data.length > 0) {
+    pbn.Data = node.data
+  } else {
+    pbn.Data = new Buffer(0)
+  }
+
+  if (node.links.length > 0) {
+    pbn.Links = node.links.map((link) => {
+      return {
+        Hash: link.multihash,
+        Name: link.name,
+        Tsize: link.size
+      }
+    })
+  } else {
+    pbn.Links = null
+  }
+
+  return pbn
 }
 
+exports.serialize = serialize
+exports.deserialize = deserialize
+exports.cid = cid
```

Comment:
@dignifiedquire go through https://github.com/ipld/interface-ipld-format and https://github.com/ipld/js-ipld-resolver/pull/60#issuecomment-254094608